### PR TITLE
Add enumerate-blocking-processes.yml

### PR DESCRIPTION
**Pull request by Ana06**
The AVOSLOCKER ransomware uses the APIs `RmRegisterResources` and [`RmGetList`](https://learn.microsoft.com/en-us/windows/win32/api/restartmanager/nf-restartmanager-rmgetlist) to get a list of processes blocking the file to encrypt and stop them.

### CODE REVIEW

File: host-interaction/process/enumerate-blocking-processes.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  meta:
+    name: enumerate blocking processes
+    namespace: host-interaction/process
+    authors:
+      - Ana06
+    scope: function
+    references:
+      - https://www.malwarebytes.com/blog/threat-intelligence/2021/07/avoslocker-enters-the-ransomware-scene-asks-for-partners
+    examples:
+      - 43b7a60c0ef8b4af001f45a0c57410b7374b1d75a6811e0dfc86e4d60f503856:0x402C5E
```
Comment (mr-tz) [inline host-interaction/process/enumerate-blocking-processes.yml:11]: Should we add this to the testfiles or remove here?

File: host-interaction/process/enumerate-blocking-processes.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+rule:
+  meta:
+    name: enumerate blocking processes
```
Comment (williballenthin) [inline host-interaction/process/enumerate-blocking-processes.yml:3]: this name isn't *super* descriptive to me: its not clear what the process is blocking. would you at least add a description that explains what this is?

File: host-interaction/process/enumerate-blocking-processes.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    namespace: host-interaction/process
+    authors:
+      - Ana06
```
Comment (williballenthin) [inline host-interaction/process/enumerate-blocking-processes.yml:6]:
```suggestion
      - "@Ana06"
```

File: host-interaction/process/enumerate-blocking-processes.yml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    scope: function
+    references:
+      - https://www.malwarebytes.com/blog/threat-intelligence/2021/07/avoslocker-enters-the-ransomware-scene-asks-for-partners
+    examples:
+      - 43b7a60c0ef8b4af001f45a0c57410b7374b1d75a6811e0dfc86e4d60f503856:0x402C5E
+  features:
+    - and:
+      - api: RmRegisterResources
+      - api: RmGetList
```
Comment (williballenthin) [inline host-interaction/process/enumerate-blocking-processes.yml:15]: recommend adding the DLL name to be explicit, but not required

File: host-interaction/process/enumerate-blocking-processes.yml (hunk 1 above)
Comment (mr-tz) [inline host-interaction/process/enumerate-blocking-processes.yml:11]:
```suggestion
    # examples:
    #  - 43b7a60c0ef8b4af001f45a0c57410b7374b1d75a6811e0dfc86e4d60f503856:0x402C5E
```

File: host-interaction/process/enumerate-blocking-processes.yml (hunk 4 above)
Comment (mr-tz) [inline host-interaction/process/enumerate-blocking-processes.yml:15]:
```suggestion
      - api: rstrtmgr.RmRegisterResources
      - api: rstrtmgr.RmGetList
```

File: host-interaction/process/enumerate-blocking-processes.yml (hunk 2 above)
Comment (mr-tz) [inline host-interaction/process/enumerate-blocking-processes.yml:3]:
```suggestion
    name: enumerate processes that use resource
```

**Comment by mr-tz** [comment]
Thanks, @Ana06!

**Pull request merged**